materialize-sql/driver.go

```diff
@@ -360,39 +352,6 @@ func (d *Driver) NewTransactor(ctx context.Context, open pm.Request_Open) (pm.Tr
 	return transactor, &pm.Response_Opened{RuntimeCheckpoint: cp}, nil
 }
 
-func isBindingMigratable(existingBinding *pf.MaterializationSpec_Binding, newBinding *pf.MaterializationSpec_Binding) bool {
```

Comment:
This function was all about making sure you can't remove keys from a materialized collection, and that any new fields are not required.

We specifically don't want to restrict new fields to only non-required ones anymore, and collection keys are always required to be materialized via constraints - see [here](https://github.com/estuary/connectors/blob/2ed68817d2f043bb34781cab47211aa3d2d46fd4/materialize-sql/validate.go#L95-L97) for new materializations and [here](https://github.com/estuary/connectors/blob/2ed68817d2f043bb34781cab47211aa3d2d46fd4/materialize-sql/validate.go#L134-L139) for modifications to existing materializations.